crypto/keymanager_test.go

```diff
@@ -0,0 +1,63 @@
+package crypto_test
+
+import (
+	"testing"
+
+	"github.com/tbd54566975/web5-go/crypto"
+	"github.com/tbd54566975/web5-go/crypto/dsa"
+)
+
+func TestGeneratePrivateKey(t *testing.T) {
+	keyManager := crypto.NewLocalKeyManager()
+
+	keyID, err := keyManager.GeneratePrivateKey(dsa.AlgorithmID.SECP256K1)
+	if err != nil {
+		t.Errorf("failed to generate private key: %v", err.Error())
+	}
+
+	if keyID == "" {
+		t.Errorf("keyID is empty")
+	}
+}
+
+func TestGetPublicKey(t *testing.T) {
+	keyManager := crypto.NewLocalKeyManager()
+
+	keyID, err := keyManager.GeneratePrivateKey(dsa.AlgorithmID.SECP256K1)
+	if err != nil {
+		t.Errorf("failed to generate private key: %v", err.Error())
+	}
+
+	publicKey, err := keyManager.GetPublicKey(keyID)
+	if err != nil {
+		t.Errorf("failed to get public key: %v", err.Error())
+	}
+
+	thumbprint, err := publicKey.ComputeThumbprint()
+	if err != nil {
+		t.Errorf("failed to compute thumbprint: %v", err.Error())
+	}
+
+	if thumbprint != keyID {
+		t.Errorf("unexpected keyID. expected: %v got: %v", keyID, thumbprint)
+	}
+}
+
+func TestSign(t *testing.T) {
+	keyManager := crypto.NewLocalKeyManager()
+
+	keyID, err := keyManager.GeneratePrivateKey(dsa.AlgorithmID.SECP256K1)
+	if err != nil {
+		t.Errorf("failed to generate private key: %v", err.Error())
+	}
+
+	payload := []byte("hello world")
+	signature, err := keyManager.Sign(keyID, payload)
+	if err != nil {
+		t.Errorf("failed to sign payload: %v", err.Error())
```

Comment:
FWIW I think writing messages for every assertion is a waste of time, though it doesn't hurt. The rationale is that, 100% of the time when a test assertion fails, you're going to look at the test code and unless the assertion is non-trivial (in which case a message would be appropriate), it will be immediately obvious what's causing the test to fail.

This is also partly why I prefer an assertion library: `assert.NoError(t, err)` will give a message about as useful as this one, with ~zero effort.

Reply:
I like this, so long as production telemetry has sufficient diagnostic information

Like, usually all that is needed is the exact source code line number where the error originates, which also implies the git commit hash for the running artifact

Reply:
Potential misunderstanding from me here https://github.com/TBD54566975/web5-go/pull/2#discussion_r1469691463

Reply:
done